ultra/ultra/utils/curriculum/coordinator.py

```diff
@@ -134,73 +170,74 @@ def graduate(self, index, average_scenarios_passed=None):
         ) == 0 and index != 0:
-            # Switch grade
-            self.next_grade()
-            self.grade_counter += 1
+            self.next_train_grade()
             self.display()
-            self.grade_checkpoints.append(index)
+            self.grade_counter += 1
             return True
         else:
             return False
 
-    def pass_based(self, index, average_scenarios_passed):
+    def pass_based(self, index, average_reached_goal) -> bool:
+        """Graduates the agent after it has obtained an
+            average reached goal greater than or equal to
+            the specified pass rate
+
+        Args:
+            index (int): reference to an episode in an sequence of episodes
+            average_reached_goal (float): mean value of episodes that the
+                                          agent has completed over some
+                                          sampling rate
+
+        Returns:
+            the eligibility of the agents to enter the next grade
+        """
         # Switch to next grade on the basis of certain percentage of completed scenarios
         if index != 0:
-            if average_scenarios_passed >= CurriculumInfo.pass_based_pass_rate:
-                print(f"({index}) AVERAGE SCENARIOS PASSED: {average_scenarios_passed}")
-                self.next_grade()
-                self.grade_counter += 1
-                # self.episode_per_grade = 0
+            if average_reached_goal >= CurriculumInfo.pass_based_pass_rate:
+                print(f"({index}) AVERAGE SCENARIOS PASSED: {average_reached_goal}")
+                self.next_train_grade()
                 self.display()
-                self.grade_checkpoints.append(index)
+                self.grade_counter += 1
                 return True
             else:
                 return False
         else:
             self.display()
-            self.grade_checkpoints.append(index)
-
-    @staticmethod
-    def calculate_average_scenario_passed(
-        episode, total_scenarios_passed, agents, asp, rate=None
-    ):
-        try:
-            sample_rate = CurriculumInfo.pass_based_sample_rate
-        except AttributeError as e:
-            sample_rate = rate
-
-        if (episode.index + 1) % sample_rate == 0:
-            total_scenarios_passed += episode.info[episode.active_tag][
-                list(agents.keys())[0]
-            ].data["reached_goal"]
-            average_scenarios_passed = total_scenarios_passed / sample_rate
-            total_scenarios_passed = 0.0
-            return average_scenarios_passed, total_scenarios_passed
-        else:
-            total_scenarios_passed += episode.info[episode.active_tag][
-                list(agents.keys())[0]
-            ].data["reached_goal"]
 
     def display(self):
+        """Prints information about succeeding grade
+
+        Raises:
+            AttributeError : Unexpected error due to self.grade_counter
+                             not being defined
```

Comment:
Does this throw an error? It looks like `self.grade_counter` is initialized in the coordinator's construction, so I think it should always be defined? Let me know your thoughts and/or what you experienced.

Reply:
@christianjans The problem was actually that the self.train_grade was not defined. The problem originated from one of the curriculum test cases not defining the train_grade. I made some changes which eliminates the use of the try-catch statement